pkg/services/ngalert/state/state.go

```diff
@@ -174,12 +174,17 @@ func (a *State) resultNoData(alertRule *models.AlertRule, result eval.Result) {
 }
 
 func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	if a.State == eval.Pending || a.State == eval.Normal && !a.Resolved {
+	if a.State == eval.Pending {
+		// We do not send notifications for pending states
 		return false
+	} else if a.State == eval.Normal {
+		// We should send a notification if the state is Normal because it was resolved
+		return a.Resolved
+	} else {
+		// We should send, and re-send notifications, each time LastSentAt is <= LastEvaluationTime + resendDelay
+		nextSent := a.LastSentAt.Add(resendDelay)
+		return nextSent.Before(a.LastEvaluationTime) || nextSent.Equal(a.LastEvaluationTime)
```

Comment:
```suggestion
    switch a.State {
	case eval.Pending:
	    // We do not send notifications for pending states
		return false
	case eval.Normal:
	    // We should send a notification if the state is Normal because it was resolved
		return a.Resolved
	default:
		// We should send, and re-send notifications, each time LastSentAt is <= LastEvaluationTime + resendDelay
		nextSent := a.LastSentAt.Add(resendDelay)
		return nextSent.Before(a.LastEvaluationTime) || nextSent.Equal(a.LastEvaluationTime)
    }
```

Reply:
I have written `NeedsSending` below with both an if statement and switch statement, what do we think?

![Screenshot 2022-09-14 at 11 53 15](https://user-images.githubusercontent.com/85952834/190135670-dcd85e1e-24c8-465c-ad5e-ad19f423bd4d.png)

![Screenshot 2022-09-14 at 11 53 51](https://user-images.githubusercontent.com/85952834/190135676-cf27df1f-79f6-4a7a-b04f-f70b98dcedb8.png)

Reply:
I have written `needsSending` below with both an if statement and a switch statement, what do we think?

```
func (a *State) NeedsSending(resendDelay time.Duration) bool {
	if a.State == eval.Pending {
		// We do not send notifications for pending states
		return false
	} else if a.State == eval.Normal {
		// We should send a notification if the state is Normal because it was resolved
		return a.Resolved
	} else {
		// We should send, and re-send notifications, each time LastSentAt is <= LastEvaluationTime + resendDelay
		nextSent := a.LastSentAt.Add(resendDelay)
		return nextSent.Before(a.LastEvaluationTime) || nextSent.Equal(a.LastEvaluationTime)
	}
}
```

```
func (a *State) NeedsSending(resendDelay time.Duration) bool {
	switch a.State {
	case eval.Pending:
		// We do not send notifications for pending states
		return false
	case eval.Normal:
		// We should send a notification if the state is Normal because it was resolved
		return a.Resolved
	default:
		// We should send, and re-send notifications, each time LastSentAt is <= LastEvaluationTime + resendDelay
		nextSent := a.LastSentAt.Add(resendDelay)
		return nextSent.Before(a.LastEvaluationTime) || nextSent.Equal(a.LastEvaluationTime)
	}
}
```

Reply:
I am more in favor of switch

Reply:
I agree with Yuri. With three cases and always comparing `a.State` the switch is easier for me to read.